deepchem/models/tests/test_reload.py

```diff
@@ -282,6 +282,42 @@ def test_robust_multitask_classification_reload():
   assert scores[classification_metric.name] > .9
 
 
+def test_normalizing_flow_model_reload():
+  """Test that RobustMultitaskRegressor can be reloaded correctly."""
+  from deepchem.models.normalizing_flows import NormalizingFlow, NormalizingFlowModel
+  import tensorflow_probability as tfp
+  tfd = tfp.distributions
+  tfb = tfp.bijectors
+  tfk = tf.keras
+  tfk.backend.set_floatx('float64')
+
+  model_dir = tempfile.mkdtemp()
+
+  Made = tfb.AutoregressiveNetwork(
+      params=2, hidden_units=[512, 512], activation='relu')
+
+  flow_layers = [tfb.MaskedAutoregressiveFlow(shift_and_log_scale_fn=Made)]
+  # 3D Multivariate Gaussian base distribution
+  nf = NormalizingFlow(
+      base_distribution=tfd.MultivariateNormalDiag(
+          loc=np.zeros(2), scale_diag=np.ones(2)),
+      flow_layers=flow_layers)
+
+  nfm = NormalizingFlowModel(nf, model_dir=model_dir)
+
+  target_distribution = tfd.MultivariateNormalDiag(loc=np.array([1., 0.]))
+  dataset = dc.data.NumpyDataset(X=target_distribution.sample(96))
+  final = nfm.fit(dataset, nb_epoch=1)
+
+  assert nfm.flow.sample().numpy().shape == (2,)
+
+  reloaded_model = NormalizingFlowModel(nf, model_dir=model_dir)
+  reloaded_model.restore()
+
+  # Check that reloaded model can sample from the distribution
+  assert reloaded_model.flow.sample().numpy().shape == (2,)
```

Comment:
For the reload test, we should make sure that `nfm` and `reloaded_model` return the exact same output after the reload is complete. Right now we're only checking shape. Could we do a direct output equality check here as well?

Reply:
Sounds good! The sampling is random so the best I can do is check shape there, but I added a test to check that density estimation returns the exact same output after the reload.